js/main.js

```diff
@@ -4,32 +4,58 @@ const $imagePlaceholder = document.querySelector('.image-placeholder');
 const entryImagePlaceholder = 'images/placeholder-image-square.jpg';
 const entryForm = document.querySelector('div[data-view="entry-form"]');
 const entries = document.querySelector('div[data-view="entries"]');
+const $entryFormText = document.querySelector('#new-entry-edit-entry');
+// change below
+const $form = document.querySelector('form');
+// bs FINE NOW
+const $idTitle = document.getElementById('new-entry-edit-entry');
 
 $photoURL.addEventListener('input', function (event) {
   $imagePlaceholder.setAttribute('src', event.target.value);
 });
 
-const $entryForm = document.querySelector('form');
-$entryForm.addEventListener('submit', function (event) {
+// const $entryForm = document.querySelector('form');
+$form.addEventListener('submit', function (event) {
   event.preventDefault();
-  const singleData = {
-    title: event.target.elements.title.value,
-    comment: event.target.elements.comment.value,
-    url: event.target.elements.url.value,
-    entryId: data.nextEntryId
-  };
-  data.nextEntryId++;
-  data.entries.unshift(singleData);
-  $imagePlaceholder.setAttribute('src', entryImagePlaceholder);
-  $entryForm.reset();
-
-  // const $renderEntry = renderEntry(singleData);
-  entriesList.prepend(renderEntry(singleData));
-  viewSwap('entries');
+  if (data.editing === null) {
+    $imagePlaceholder.setAttribute('src', entryImagePlaceholder);
+    const singleData = {
+      title: $form.elements.title.value,
+      comment: $form.elements.notes.value,
+      url: $form.elements.url.value,
+      entryId: data.nextEntryId
+    };
+    data.nextEntryId++;
+    data.entries.unshift(singleData);
+    $imagePlaceholder.setAttribute('src', entryImagePlaceholder);
+    $form.reset();
+
+    // const $renderEntry = renderEntry(singleData);
+    entriesList.prepend(renderEntry(singleData));
+    viewSwap('entries');
+  } else {
+    data.editing.title = event.target.elements.title.value;
+    data.editing.comment = event.target.elements.notes.value;
+    data.editing.url = event.target.elements.url.value;
```

Comment:
`data.editing` already has these values, so you shouldn't be updating `data.editing`, you want to be updating `data.entries`. So you'll need to find the current entry that exists in `data.entries` and replace it with these newly edited values